src/SonarScanner.MSBuild.Common/StringExtensions.cs

```diff
@@ -40,4 +40,18 @@ public static class StringExtensions
     public static string ReplaceCaseInsensitive(this string input, string oldValue, string newValue) =>
         // Based on https://stackoverflow.com/a/6276029/7156760
         Regex.Replace(input, Regex.Escape(oldValue), newValue.Replace("$", "$$"), RegexOptions.IgnoreCase, RegexConstants.DefaultTimeout);
+
+    public static string RedactSensitiveData(this string input)
+    {
+        var indexes = SonarProperties.SensitivePropertyKeys
```

Comment:
This method is performance sensitive on the log output (e.g. React apps will have millions of DEBUG entries). Consider writing this allocation free by unrolling the LINQ query. e.g.:
```cs
var min = default(int?);
foreach(var key in SonarProperties.SensitivePropertyKeys)
{
   var current = input.IndexOf(key, StringComparison.OrdinalIgnoreCase);
   if (current != -1)
   {
      min = Math.Min(min ?? current, current);
   }
}
```

Add a comment about the sensitivity of the method.